pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -266,12 +280,38 @@ func (m *AddOnManager) verifyRequiredVariablesInContext(context *command.Executi
 
 	if len(missingVars) > 0 {
 		missing := strings.TrimSpace(strings.Join(missingVars, ", "))
-		return errors.New(fmt.Sprintf("The variable(s) %s are required by the add-on, but are not defined in the context.", missing))
+		return fmt.Errorf("The variable(s) %s are required by the add-on, but are not defined in the context", missing)
 	}
 
 	return nil
 }
 
+func (m *AddOnManager) verifyRequiredOpenshiftVersion(context *command.ExecutionContext, meta addon.AddOnMeta) error {
+	dockerCommander := context.GetDockerCommander()
+	versionInfo, err := dockerCommander.Exec(" ", constants.OpenshiftContainerName, "openshift", "version")
+	if err != nil {
+		return err
+	}
+	openShiftVersionAlongWithCommitSha := strings.Split(strings.Split(versionInfo, "\n")[0], " ")[1]
```

Comment:
These lines are a bit of black magic to me. Would a regexp not be easier than all the splits? Some comment on what the format is you are going to process would be nice as well. I think I'd extract the lines which extract the OpenShift version into a helper function which can easily be tested and documents the expected inputs as a side effect. I am afraid this code is fragfile and hard to maintain.